framework/src/meshgenerators/FlipSidesetGenerator.C

```diff
@@ -0,0 +1,87 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FlipSidesetGenerator.h"
+
+#include "CastUniquePointer.h"
+
+registerMooseObject("MooseApp", FlipSidesetGenerator);
+
+InputParameters
+FlipSidesetGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+  params.addClassDescription("A Mesh Generator which flips a given sideset");
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<BoundaryName>("boundary", "The sideset (boundary) that will be flipped");
+  return params;
+}
+
+FlipSidesetGenerator::FlipSidesetGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _input(getMesh("input")),
+    _sideset_name(getParam<BoundaryName>("boundary"))
+{
+}
+
+std::unique_ptr<MeshBase>
+FlipSidesetGenerator::generate()
+{
+  //get boundary info
+  BoundaryInfo & boundary_info = _input->get_boundary_info();
+  //get id of the input sideset
+  boundary_id_type sideset_id = boundary_info.get_id_by_name(_sideset_name);
+  
+  //Throw an error if the sideset doesn't exist
+  if (sideset_id == libMesh::BoundaryInfo::invalid_id)
+    mooseError("sideset doesn't exist in mesh");
+  
+  //get sideset map
+  std::multimap<const Elem *, std::pair<unsigned short int, boundary_id_type>> sideset_map = boundary_info.get_sideset_map();
+
+  //old_elem is the original element attached to the sideset before flipping
+  //new_elem is the element attached to the sideset after flipping
+  for(const auto & [old_elem, id_pair] : sideset_map)
+  {
+    boundary_id_type boundary_id = std::get<1>(id_pair);
+    if (boundary_id == sideset_id)
+    {
+      unsigned short int old_side_id = std::get<0>(id_pair);
+      dof_id_type old_elem_id = old_elem->id();
+      const Elem * new_elem = old_elem->neighbor_ptr(old_side_id);
+      
+      //Throw an error if the old element doesn't have a neighbor on the old side
+      if (!new_elem)
+        mooseError("elem " + std::to_string(old_elem_id) + " doesn't have a neighbor through side " + 
+                    std::to_string(old_side_id) + " therefore it can't be flipped");
+
+      unsigned int new_elem_n_sides = new_elem->n_sides();
+      unsigned int new_side_id;
+      
+      //To find the new_side_id:
+      //loop over new_elem sides until you find the neighbor which have the same id as the old elem
+      for (unsigned int i = 0; i < new_elem_n_sides; ++i)
```

Comment:
Did you ever try that "neighbor through face" idea? Which would get this information from the mesh topology instead of looping through and checking each face as a candidate?

Reply:
Do you mean using a built in function that takes a neighbor and returns the side?

Reply:
Yes, I thought you had found a possible candidate - I'd prefer that if possible since it'd be a bit more readable code and would hopefully be faster (not a constraint from the test cases you have, but could be important for larger meshes).

Reply:
The only function I found that seemed to do something similar was: which_side_am_i	(const Elem * e) which we discussed before and found that is not doing what I thought, other than this I didn't find a suitable candidate function

Reply:
Hmm, I am not sure - ok let's leave the code as-is for now, and we can ask on the MOOSE repo if they have any advice on where to look for something like that (if it exists).

Reply:
Can you check which_neighbor_am_i()
I think this may be the one, I didn't understand its documentation the first time but may be it the correct one
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1Elem.html#ac4fd20bbdc7f49f25ecc67d67f2a2295

Reply:
I think that might be it (not 100% sure until we test it). It looks like these lines in `which_neighbor_am_i` look like they're doing the same looping-over-candidates you are doing (but it would be preferred if we can use the `which_neighbor_am_i` to reduce code duplication).

```
2326   for (unsigned int s=0, n_s = this->[n_sides]; s != n_s; ++s)
 2327     if (this->[neighbor_ptr](s) == eparent)
 2328       return s;
```

Can you try it out?